### ISSUE TITLE
new Show Script insufficient to be usable

### ISSUE THREAD

**Issue by Zimmi48** [Zimmi48/bugzilla-test#2820]
Note: the issue was created automatically with bugzilla2github tool

# Bugzilla Bug ID: 2820
Date: 2012-06-18 12:21:03 +0200
From: @_roglo
To:   
CC:   @_aspiwack, @_letouzey

Last updated: 2012-07-11 20:45:09 +0200

## Bugzilla Comment ID: 5164
Date: 2012-06-18 12:21:03 +0200
From: @_roglo

Hi there!

Really, I cannot use 8.4 because the new Show Script (even if you made an effort to make a new version of it) is not usable for me because not compatible enough with before.

I know that almost everybody uses proofgeneral or coqide, but I prefer not using them (it is my choice) and restrict myself to the usage of the toplevel command "coqtop", using my favorite text editor, and cuts-and-pastes. I consider that imposing people to use embedding programs (proofgeneral, coqide) is bad policy, and prevent future possible programming of other kinds of interfaces. Moreover, having the possibility to use toplevel commands and only them is the minimum.

The new Show Script lacks:
- indentation like before (or any new consistent one)
- displaying the current script after "Load File" when a tactics happens do fail in the middle of a proof (it displays nothing).

Is it planned to have a real Show Script? For me, it is really blocking. I am still in version 8.3 because of that.

## Bugzilla Comment ID: 5226
Date: 2012-06-22 10:59:34 +0200
From: @_aspiwack

There are two different things here. I'm afraid they're both difficult, and unlikely to be solved quickly. We understand how inconvenient it is to you, but we try to move forward and it occasionally means breaking things for some users.

The indentation previously relied on information that is just not there anymore (it's a byproduct of the changes of the proof engine, in which these informations are hard to retrieve. Mostly because they don't quite make sense there). It is not to say nothing can be done. But we don't know how to get out of there yet.

The other issue pertains to the change of semantics of Load. Previously it would just read the file line by line. It has now a more static semantics, where it tries to read the whole file, and failure cancels the whole operation. The point is that it now makes sense to backtrack on a Load, and it can hence safely be used in Coqide. Maybe we could try to display more information when a Load fails, but I don't know what would be relevant.

## Bugzilla Comment ID: 5285
Date: 2012-07-07 20:30:51 +0200
From: @_letouzey

First issue (indentation) should be solved now : I've integrated in trunk
and 8.4 a refined version of the patch proposed by Daniel on coq-dev.

Second issue (Show Script across Load) : I'll see want I can do, but I make
no promise. As said by Arnaud, this isn't straightforward with the new
backtracking framework.

Best,

Pierre

## Bugzilla Comment ID: 5295
Date: 2012-07-11 20:45:09 +0200
From: @_letouzey

Temporary fix for the second part of the bug (Show Script after Load) :
A flag now allows to tweak the behavior of Load with respect to the
history stack on which Show Script is based.

In short, "Unset Atomic Load" should make Show Script work across Load
in 8.4 and trunk.

Caveat : do not use this setting in CoqIDE (and possibly ProofGeneral),
otherwise backtracking over a Load will fail.

This fix is rather sub-optimal, I'll try to improve it later.

Best,
Pierre
